language/move-compiler/src/typing/core.rs

```diff
@@ -1703,3 +1728,23 @@ fn check_num_tvar_(subst: &Subst, loc: Loc, ty: &Type) -> bool {
         _ => false,
     }
 }
+//**************************************************************************************************
+// Function type check
+//**************************************************************************************************
+
+pub fn check_non_fun(context: &mut Context, ty: &Type) {
```

Comment:
I think this will need to be moved into `typing/translate` depending on the usages

Reply:
Looking at more usages maybe its fine.. still trying to wrap my head around this